### PR TITLE
chore: Use VHD template for containerD and docker Windows VHD build

### PR DESCRIPTION
**Pull request by mainred**
Changes included in this PR:
- extract the common code for building windows VHD
- create a new pipeline yaml file, which will be used in VHD publishing **for production**, in which [azure DevOps building parameters](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/runtime-parameters?view=azure-devops&tabs=script) dynamically select stages.
-  use official building image for Windows VHD build `mcr.microsoft.com/oss/azcu/go-dev:v1.28.5`
This change has been validated in this pipeline https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=38988835&view=results

NOTE:
- current change won't break anything, we have only enabled a pipeline for test. Also the pipeline shows in the check-in because I manually triggered a building.
- We don't in this PR, but will, change the check-in pipeline before we officially build containerd-supported windows VHD

### CODE REVIEW

File: .pipelines/templates/.builder-release-template-windows.yaml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      -e IMAGE_VERSION=${WINDOWS_VERSION} \
+      -e HYPERV_GENERATION="V1" \
+      ${CONTAINER_IMAGE} make -f packer.mk generate-sas
+    displayName: Getting Shared Access Signature URI
+    condition: and(eq(variables.DRY_RUN, 'False'), eq(variables['MODE'], 'default'))
+  - task: PublishPipelineArtifact@1
+    inputs:
+        artifactName: 'publishing-info-${{ parameters.artifactName }}'
+        targetPath: 'vhd-publishing-info.json'
+    condition: and(eq(variables.DRY_RUN, 'False'), ne(variables['MODE'], 'sigMode'))
```
Comment (AbelHu) [inline .pipelines/templates/.builder-release-template-windows.yaml:139]: Is this for PIR? We also need to support SIG for production soon.

Comment (mainred) [reply .pipelines/templates/.builder-release-template-windows.yaml:139]: SIG mode means people build images to SIG by specifying sig name and image definition etc. sigMode is not for production.

Comment (AbelHu) [reply .pipelines/templates/.builder-release-template-windows.yaml:139]: So is it not used for AKS SIG?

Comment (mainred) [reply .pipelines/templates/.builder-release-template-windows.yaml:139]: AKS SIG is outside of building VHD. This sigMod is to bring your own SIG image for testing.

File: .pipelines/.vsts-vhd-builder-release-windows.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          echo '##vso[task.setvariable variable=HYPERV_GENERATION]V1'
+          echo '##vso[task.setvariable variable=AZURE_VM_SIZE]Standard_D4s_v3'
+          echo '##vso[task.setvariable variable=CONTAINER_RUNTIME]docker'
+          echo '##vso[task.setvariable variable=WINDOWS_SKU]2019'
+        displayName: Setup Build Variables
+      - template: ./templates/.builder-release-template-windows.yaml
+        parameters:
+          artifactName: 2019
+
+  - phase: build_vhd_2019_containerd
```
Comment (mainred) [inline .pipelines/.vsts-vhd-builder-release-windows.yaml:39]: Just a reminder: we don't need to cache so many k8s versions for containerD case.

**Review by AbelHu — APPROVED** [review]

**Pull request merged**